python_configure.bzl

```diff
@@ -244,26 +244,55 @@ def _check_python_bin(repository_ctx, python_bin):
             python_bin,
         ))
 
-def _get_python_include(repository_ctx, python_bin):
-    """Gets the python include path."""
-    python_lib = repository_ctx.os.environ.get(_PYTHON_LIB_PATH)
-    if python_lib != None:
-        return python_lib
+def _get_python_version(repository_ctx, python_bin):
+    """Returns a tuple (major_version, minor_version)."""
     result = _execute(
         repository_ctx,
         [
             python_bin,
             "-c",
-            "from __future__ import print_function;" +
-            "from distutils import sysconfig;" +
-            "print(sysconfig.get_python_inc())",
+            'import sys; print(str(sys.version_info[0]) + "." + str(sys.version_info[1]))',
         ],
-        error_msg = "Problem getting python include path.",
-        error_details = ("Is the Python binary path set up right? " +
-                         "(See ./configure or " + _PYTHON_BIN_PATH + ".) " +
-                         "Is distutils installed?"),
+        error_msg = "Problem getting Python version using binary " + python_bin,
+        error_details = "",
     )
-    return result.stdout.splitlines()[0]
+    version = result.stdout.splitlines()[0].split(".")
+    version = (int(version[0]), int(version[1]))
+    return version
+
+def _get_python_include(repository_ctx, python_bin):
+    """Gets the python include path."""
+    version = _get_python_version(repository_ctx, python_bin)
+    if version >= (3, 10):
+        result = _execute(
+            repository_ctx,
+            [
+                python_bin,
+                "-c",
+                "import sysconfig; print(sysconfig.get_path('include'))",
+            ],
+            error_msg = "Problem getting python include path for Python version " + str(version),
+            error_details = ("Is the Python binary path set up right? " +
+                             "(See ./configure or " + _PYTHON_BIN_PATH + ".) " +
+                             "Is distutils installed?"),
+        )
+        return result.stdout.splitlines()[0]
+    else:
+        result = _execute(
```

Comment:
I just looked:

https://docs.python.org/3/library/sysconfig.html

sysconfig already existed in Python 3.2. pybind11 only works for Python 3.6 or newer. My bet is that this old code can be completely deleted.

My vote: just delete it straightaway, in the very unlikely case that someone complains, we can still add it back.

Another vote: I'd keep the new `_get_python_version()` function even if it's not used. It'll probably be handy later.